services/components/lifecycle/bot_logger_test.go

```diff
@@ -0,0 +1,200 @@
+		defaultAgentLogAvgMaxCharsPerLine*defaultAgentLogTailLines,
+	).Return("some log", nil).Times(1)
+
+	s.botClient.EXPECT().LoadBotContainers(ctx).Return(mockContainers, nil)
+	botLogger.SendBotLogs(ctx)
+}
+
+// should fail if there is an error loading
+// bot containers
+func (s *BotLoggerSuite) TestLoadBotContainersError() {
+	botLogger := NewBotLogger(
+		s.botClient, s.dockerClient, s.key,
+		func(agents agentlogs.Agents, authToken string) error {
+			return nil
+		},
+	)
+	ctx := context.Background()
+
+	mockContainers := []types.Container{}
+
+	s.botClient.EXPECT().LoadBotContainers(ctx).Return(mockContainers, errors.New("test"))
+	s.r.Error(botLogger.SendBotLogs(ctx), "failed to load the bot containers: test")
+}
+
+// Should not send agent logs if fails
+// to get container logs but continue processing
+func (s *BotLoggerSuite) TestGetContainerLogsError() {
+	botLogger := NewBotLogger(
+		s.botClient, s.dockerClient, s.key,
+		func(agents agentlogs.Agents, authToken string) error {
+			s.r.Equal(1, len(agents))
+			s.r.Equal("bot2ID", agents[0].ID)
+			s.r.Equal("some log", agents[0].Logs)
+			return nil
+		},
+	)
+	ctx := context.Background()
+
+	mockContainers := []types.Container{
+		{
+			ID:    "bot1",
+			Image: "forta/bot:latest",
+			Labels: map[string]string{
+				docker.LabelFortaSettingsAgentLogsEnable: "true",
+			},
+		},
+		{
+			ID:    "bot2",
+			Image: "forta/bot:latest",
+			Labels: map[string]string{
+				docker.LabelFortaSettingsAgentLogsEnable: "true",
+				docker.LabelFortaBotID:                   "bot2ID",
+			},
+		},
+	}
+
+	s.botClient.EXPECT().LoadBotContainers(ctx).Return(mockContainers, nil)
+
+	s.dockerClient.EXPECT().GetContainerLogs(
+		ctx, "bot1",
+		strconv.Itoa(defaultAgentLogTailLines),
+		defaultAgentLogAvgMaxCharsPerLine*defaultAgentLogTailLines,
+	).Return("", errors.New("test")).Times(1)
+
+	s.dockerClient.EXPECT().GetContainerLogs(
+		ctx, "bot2",
+		strconv.Itoa(defaultAgentLogTailLines),
+		defaultAgentLogAvgMaxCharsPerLine*defaultAgentLogTailLines,
+	).Return("some log", nil).Times(1)
+
+	botLogger.SendBotLogs(ctx)
+}
+
+// Fails sending agent logs
+func (s *BotLoggerSuite) TestFailsToSendLogs() {
+	botLogger := NewBotLogger(
+		s.botClient, s.dockerClient, s.key,
+		func(agents agentlogs.Agents, authToken string) error {
+			return errors.New("test")
+		},
+	)
+	ctx := context.Background()
+
+	mockContainers := []types.Container{
+		{
+			ID:    "bot1",
+			Image: "forta/bot:latest",
+			Labels: map[string]string{
+				docker.LabelFortaSettingsAgentLogsEnable: "true",
+				docker.LabelFortaBotID:                   "bot1ID",
+			},
+		},
+	}
+
+	s.botClient.EXPECT().LoadBotContainers(ctx).Return(mockContainers, nil)
+
+	s.dockerClient.EXPECT().GetContainerLogs(
+		ctx, "bot1",
+		strconv.Itoa(defaultAgentLogTailLines),
+		defaultAgentLogAvgMaxCharsPerLine*defaultAgentLogTailLines,
+	).Return("some log", nil).Times(1)
+
+	s.r.Error(botLogger.SendBotLogs(ctx), "failed to send agent logs: test")
```

Comment:
I think we might need to use `EqualError` or `ErrorContains` instead of `Error` here. If we want to only assert that there is a non-nil error, then `Error` is more useful.